src/gf.c

```diff
@@ -2942,6 +2942,19 @@ JL_DLLEXPORT int jl_compile_hint(jl_tupletype_t *types)
     return 1;
 }
 
+// true if jl_get_compile_hint_specialization can find a method instance for us to try
+// to compile
+JL_DLLEXPORT int jl_has_compile_hint_specialization(jl_tupletype_t *types)
+{
+    size_t world = jl_atomic_load_acquire(&jl_world_counter);
+    size_t min_valid = 0;
+    size_t max_valid = ~(size_t)0;
+    jl_method_instance_t *mi = jl_get_compile_hint_specialization(types, world, &min_valid, &max_valid, 1);
```

Comment:
`jl_get_compile_hint_specialization` first finds a "compilable method match" then calls `jl_method_match_to_mi` to create a "method instance" -- if that step can't return `NULL` then i suppose we don't need to spend the time creating the "method instance" and could just check if we found a "method match" ?

Reply:
yeah interesting. i'm not sure whether that step is generally nullable or not. 🤔 

The code you have here, now, roughly matches what i had in mind when you described this idea to me 👍

Reply:
@kpamnany is going to ask Gabriel about this